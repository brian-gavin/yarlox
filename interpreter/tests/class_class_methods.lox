class Math {
    class square(n) {
        return n * n;
    }
    class TAU {
        return 2 * PI;
    }
    class circleArea(radius) {
        return Math.TAU * Math.square(radius);
    }
}

print Math.square(2);
print Math.TAU;
print Math.circleArea(2);