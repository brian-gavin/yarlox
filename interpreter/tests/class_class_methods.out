4
6.283185307179586
25.132741228718345
